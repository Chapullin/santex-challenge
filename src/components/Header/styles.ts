import styled from "styled-components";

export const FloatNav = styled.header`
    padding: 15px;
    position: fixed;
    width: 100%;
    display: inline;
    top: 0;
`;

export const Price = styled.div`
    color: white;
    font-weight: bold;
    float: right;
    margin-right: 30px;
`;
