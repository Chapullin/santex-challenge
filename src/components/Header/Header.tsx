import { useStateWithStorage } from "../../hooks/useStateWithStorage";
import { FloatNav, Price } from "./styles";

export function Header() {
    const { subTotal } = useStateWithStorage();

    return (
    <FloatNav style={{ background: 'red' }}>
      <img
        src="https://santex.wpengine.com/wp-content/uploads/2019/02/logo-santex@3x.png"
        alt="logo"
      />
      <Price>AR$ { subTotal }</Price>
    </FloatNav>
  );
}
