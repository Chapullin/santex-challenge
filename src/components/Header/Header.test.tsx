import { render, waitFor, screen } from "@testing-library/react";
import React from 'react';

import { Header } from "./Header";

describe("Header", () => {
    it("renders Header with img and price subTotal", async () => {
        render(<Header />);

        const { getByText } = screen;
        const displayedImage = document.querySelector("img") as HTMLImageElement;

        await waitFor(() => {
            expect(displayedImage.src).toContain("https://santex.wpengine.com/wp-content/uploads/2019/02/logo-santex@3x.png");
            expect(getByText(/0/)).toBeInTheDocument();
        });
    });

    it("should match snapshot Header", async () => {
        const { container } = render(<Header />);
        await waitFor(() => {
            expect(container.firstChild).toMatchSnapshot();
        });
    });
});
