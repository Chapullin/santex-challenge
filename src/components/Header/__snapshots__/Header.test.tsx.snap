// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot Header 1`] = `
<header
  class="sc-bcXHqe fJphvG"
  style="background: red;"
>
  <img
    alt="logo"
    src="https://santex.wpengine.com/wp-content/uploads/2019/02/logo-santex@3x.png"
  />
  <div
    class="sc-gswNZR hmTBPx"
  >
    AR$ 
    0
  </div>
</header>
`;
